core/controlplane/config/config.go

```diff
@@ -435,6 +438,7 @@ type DeploymentSettings struct {
 	CloudWatchLogging       `yaml:"cloudWatchLogging,omitempty"`
 	AmazonSsmAgent          `yaml:"amazonSsmAgent,omitempty"`
 	CloudFormationStreaming bool `yaml:"cloudFormationStreaming,omitempty"`
+	KubeDnsMasq             `yaml:"kubeDnsMasq,omitempty"`
```

Comment:
It sounds awesome! I like yours more than mine :+1:
If I could add one thing, `kubeDns.nodeLocalResolver` should be an object rather than a bool so that we can differentiate configs(like cache size, which isn't configurable right now) between node local dnsmasq and cluster-wide dnsmasq?
The node local resolver could be enabled via `kubeDns.nodeLocalResolver.enabled` then.